### ISSUE TITLE
Deprecation of componentsScan and mount

### ISSUE THREAD

**Issue by jpc-dnr** [TypedProject/ts-express-decorators#624]
## Information

- **Version:** 5.21.1
- **Type:** Issue

## Description

Since 5.21.1, properties IServerSettings.componentsScan and IServerSettings.mount are tagged deprecated but I did not find any replacement in documentation. All examples I saw still continue to reference these properties.

Did I miss something? What should be used in replacement?

Note that if I just ignore deprecated errors from TsLint, the server has no mounted routes.
Also if I downgrade Ts.ED to v5.21.0 it works fine.

Thanks!

## Example

```typescript
import { ServerLoader, ServerSettings, IServerSettings } from '@tsed/common';

@ServerSettings({}) // configuration created dynamically in $onInit
export class MyServer extends ServerLoader {
  private _serverSettings: IServerSettings;

  public async $onInit(): Promise<any> {
    this._serverSettings = {...};
    this._serverSettings.componentsScan = [...];
    this._serverSettings.mount = {...};
    this.setSettings(this._serverSettings);
  }
}
```

**Comment by Romakita** [comment]
Hello @jpc-dnr 
Method to register mount and componentScan configuration with method are deprecated.

You have to set your configuration from decorator or on contructor level:

```typescript
@ServerSettings({}) // configuration created dynamically in $onInit
export class MyServer extends ServerLoader {
  contructor(settings = {}) {
     super({
          ...settings,
          mount: [],
          componentScan: []
      }) // Work with latest version 5.27

     // or
     this.settings.mount = []
     this.settings.componentsScan = []
  }
}

// index.ts
async function bootstrap() {
    const settings = {}
    const server = await ServerLoader.bootstrap(Server, settings)
    
    await server.listen()
}

bootstrap()
```

See you
Romain

**Comment by jpc-dnr** [comment]
Thanks for your quick answer. This appears to work like a charm... at least in version 5.26.0.
Starting at 5.27.0, I'm having DI issue that seems to be unrelated to this subject.
I will first try to investigate and, if necessary, I will open another issue.

**Comment by Romakita** [comment]
Ok :) le me know if you have problems with DI in 5.27.0 
I fixed some small issues in 5.27.4 about DI 

See you

**Comment by jpc-dnr** [comment]
Thanks a lot for your help!
This issue is closed for me, I just opened another issue for the DI problem (see #626).
